### PR TITLE
Test switch

### PR DESCRIPTION
**Pull request by Amet015**

### CODE REVIEW

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java [hunk 1]
```diff
@@ -9,13 +9,14 @@
...
 
+import salesforce.utils.UtilSalesforce;
 import org.openqa.selenium.By;
 import org.openqa.selenium.JavascriptExecutor;
 import org.openqa.selenium.Keys;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
 import org.openqa.selenium.support.ui.ExpectedConditions;
-import java.util.Set;
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java]: delete this line please

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java [hunk 2]
```diff
@@ -39,9 +40,13 @@
     @FindBy(css = "input[title='Edit']")
     private WebElement editButton;
 
+    @FindBy(name = "resultsFrame")
+    private WebElement resultsFrame;
+
     @FindBy(css = "input[value='Attach File']")
     private WebElement attachFileButton;
 
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java]: delete this line please

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java [hunk 3]
```diff
@@ -75,21 +78,24 @@ protected String getCampaignName() {
...
+            webDriver.switchTo().window(parentHandle);
         }
-        // Perform the actions on new window
+    }
+
+    /**
+     * Clicks to campaign selected.
+     * @param campaignName string value.
+     */
+    private void clickCampaignSelected(final String campaignName) {
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java]: click**On**SelectedCampaign

File: src/main/java/salesforce/utils/UtilSalesforce.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Jalasoft.
+ */
+
+package salesforce.utils;
+
+import core.selenium.WebDriverManager;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/utils/UtilSalesforce.java]: delete this line

File: src/main/java/salesforce/utils/UtilSalesforce.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+ * @version 1.0 23 March 2020.
+ */
+public class UtilSalesforce {
+
+    /**
+     * Switches to new windows.
+     * @param parentHandle string value.
+     */
+    public static void switchToNewWindow(final String parentHandle) {
+        WebDriverManager.getInstance().getWebDriverWait().until(ExpectedConditions.numberOfWindowsToBe(2));
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/utils/UtilSalesforce.java]: avoid magic number **2**

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java (hunk 3 above)
Comment (Amet015) [reply src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java]: it is passive, i will change OnCampaignSelected

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java [hunk 6]
```diff
@@ -16,7 +16,8 @@
 import org.openqa.selenium.support.FindBy;
 import org.openqa.selenium.support.ui.ExpectedConditions;
 import salesforce.ui.pages.owner.OwnerEditClassicPage;
-import java.util.Set;
+import salesforce.utils.UtilSalesforce;
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java:20]: delete this line please

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java [hunk 7]
```diff
@@ -40,12 +40,15 @@
...
 
+    @FindBy(name = "resultsFrame")
+    private WebElement resultsFrame;
     @FindBy(xpath = "//a[contains(text(),'[Change]')]")
     private WebElement changeLinkText;
 
     @FindBy(css = "div[id='opp1_ileinner'] a[id]")
     private WebElement ownerLabel;
 
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java]: delete this line please

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java [hunk 8]
```diff
@@ -82,21 +83,24 @@ protected String getCampaignName() {
...
+            webDriverWait.until(ExpectedConditions.frameToBeAvailableAndSwitchToIt(resultsFrame));
+            clickOnCampaignSelected(campaignName);
+        } finally {
+            webDriver.switchTo().window(parentHandle);
         }
-        // Perform the actions on new window
+    }
+
+    /**
+     * Clicks on campaign selected.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java:97]: selected campaign

File: src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java [hunk 9]
```diff
@@ -40,12 +40,15 @@
     @FindBy(css = "input[title='Edit']")
     private WebElement editButton;
 
+    @FindBy(name = "resultsFrame")
+    private WebElement resultsFrame;
     @FindBy(xpath = "//a[contains(text(),'[Change]')]")
```
Comment (JuanCarlosMartinez4) [inline src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java]: separete this lines please

**Review by JuanCarlosMartinez4 — APPROVED** [review]

**Pull request merged**